src/wp-admin/includes/plugin.php

```diff
@@ -1476,6 +1476,24 @@ function add_submenu_page( $parent_slug, $page_title, $menu_title, $capability,
 	global $submenu, $menu, $_wp_real_parent_file, $_wp_submenu_nopriv,
 		$_registered_pages, $_parent_pages;
 
+	/*
+	 * Check that the slug parameters are strings to prevent type errors with
+	 * plugin_basename(). The error message refers to the first five parameters
+	 * in case any of the other three might need validation in the future.
+	 */
+	if ( ! is_string( $parent_slug ) || ! is_string( $menu_slug ) ) {
+		_doing_it_wrong(
+			__FUNCTION__,
+			sprintf(
+				/* translators: %s: add_submenu_page() */
+				__( 'The first five parameters passed to %s should be strings.' ),
+				'<code>add_submenu_page()</code>'
+			),
+			'6.6.0'
+		);
+		return false;
```

Comment:
I could remove the `return` to let the function continue, in case it is caused by the `null` `$parent_slug` hack that _might_ work today. At least when people see the `functions.php` deprecation notices, they could have a clearer idea that the notices were not caused by Core.

I'm not eager to create a new patch for this ticket, but another possibility is to check the `$parent_slug` specifically for `null`:
```
	if ( null === $parent_slug ) {
		_doing_it_wrong( ... );
		$parent_slug = '';
	}
```
(not as thorough but probably would catch a large majority of the errors, and the empty string might still act unexpectedly)